### ISSUE TITLE
Not able to copy objects between s3 buckets in different regions.

### ISSUE THREAD

**Issue by RajBalakrishnan** [cloudyr/aws.s3#247]
Please specify whether your issue is about:

 - [ ] a possible bug
 - [ ] a question about package functionality
 - [ ] a suggested code or documentation change, improvement to the code, or feature request

If you are reporting (1) a bug or (2) a question about code, please supply:

 - [a fully reproducible example](http://stackoverflow.com/questions/5963269/how-to-make-a-great-r-reproducible-example) using a publicly available dataset (or provide your data)
 - if an error is occurring, include the output of `traceback()` run immediately after the error occurs
 - the output of `sessionInfo()`

Put your code here:

```R
## load package
library("aws.s3")

## code goes here


## session info for your system
sessionInfo()
```

**Comment by leeper** [comment]
Can you show some code and the error it produces?

**Comment by RajBalakrishnan** [comment]
library(aws.s3)
Sys.setenv("AWS_ACCESS_KEY_ID" = "BucketA-Key_ID",
           "AWS_SECRET_ACCESS_KEY" = "BucketA-SecretKey",
           "AWS_DEFAULT_REGION" = "us-east-1")
copy_object("FileX", "FileX","BucketA(us-east-1)", "BucketB(us-west-1"))

Getting the obvious access denied error for accessing BucketB as the Sys.setenv() contains credentials for BuvketA only.

**Comment by leeper** [comment]
You need to use credentials that give you access to both buckets.

**Comment by RajBalakrishnan** [comment]
Is it possible to provide two set of credentials in Sys.setenv()?

**Comment by leeper** [comment]
No. That's an Amazon-level issue - a copy is a single AWS-side operation so you need credentials that work on both buckets. If you want to save/get locally and put in another bucket, you could use one set of credentials for the get/save and a different set for the put.
